# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1464855005 -36000
#      Thu Jun 02 18:10:05 2016 +1000
# Branch stable
# Node ID 1b5285c20e0e2ed7fc7bd3cc8a1d52e3e80e6daa
# Parent  4950810ce6058ff41b5057476210e587fccaef17
Create valid gnuplot commands even for single-entry colormaps (bug #48083).

* __go_draw_axes__.m: If the colourmap is a singleton,
  then replace it by one with two identical colours.

diff --git a/scripts/plot/util/private/__go_draw_axes__.m b/scripts/plot/util/private/__go_draw_axes__.m
--- a/scripts/plot/util/private/__go_draw_axes__.m
+++ b/scripts/plot/util/private/__go_draw_axes__.m
@@ -1627,16 +1627,20 @@ function __go_draw_axes__ (h, plot_strea
              __do_enhanced_option__ (enhanced, hlgnd));
   else
     fputs (plot_stream, "unset key;\n");
   endif
   fputs (plot_stream, "set style data lines;\n");
 
   cmap = [cmap; addedcmap];
   cmap_sz = cmap_sz + rows (addedcmap);
+  if (cmap_sz == 1)        # bug #48083, illegal one-element colormap
+    cmap = [cmap; cmap];
+    cmap_sz = 2;
+  endif
   if (mono == false && length (cmap) > 0)
     fprintf (plot_stream,
              "set palette positive color model RGB maxcolors %i;\n",
              cmap_sz);
     fprintf (plot_stream,
              "set palette file \"-\" binary record=%d using 1:2:3:4;\n",
              cmap_sz);
     fwrite (plot_stream, [1:cmap_sz; cmap.'], "float32");
